server/handlers/recovery_handler.go

```diff
@@ -0,0 +1,34 @@
+package handlers
+
+import (
+	"fmt"
+	"runtime/debug"
+)
+
+func goWithRecovery(name string, doStart func(), isQuitting func() bool, logError func(msg string, keyValuePairs ...any)) {
+	var doRecoverableStart func()
+
+	// doRecover is a helper function to recover from panics and restart the goroutine.
```

Comment:
This logic is largely lifted out of https://github.com/mattermost/mattermost/blob/5e94af13022d16dd29c079bc2a68c84a5bac6e9e/server/channels/app/platform/web_hub.go#L370-L375, except generalized (and unit tested).